nixos/modules/services/web-apps/kanboard.nix

```
@@ -0,0 +1,137 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.services.kanboard;
+
+  toStringAttrs = attrs: lib.attrsets.mapAttrs (k: v: toString v) attrs;
+in
+{
+  meta.maintainers = with lib.maintainers; [ yzx9 ];
+
+  options.services.kanboard = with lib; {
+    enable = mkEnableOption "Kanboard";
+
+    package = mkPackageOption pkgs "kanboard" { };
+
+    dataDir = mkOption {
+      type = types.str;
+      default = "/var/lib/kanboard";
+      description = "Default data folder for Kanboard.";
+      example = "/mnt/kanboard";
+    };
+
+    user = mkOption {
+      type = types.str;
+      default = "kanboard";
+      description = "User under which Kanboard runs.";
+    };
+
+    settings = mkOption {
+      type =
+        with types;
+        attrsOf (oneOf [
+          str
+          int
+          bool
+        ]);
+
+      default = { };
+
+      description = ''
+        Customize the default settings, refer to <https://github.com/kanboard/kanboard/blob/main/config.default.php>
+        for details on supported values.
+      '';
+    };
+
+    # Nginx
+    virtualHost = mkOption {
+      type = lib.types.str;
+      description = ''
+        FQDN for the kanboard instance.
+      '';
+    };
```

```suggestion
    domain = lib.mkOption {
      type = lib.types.str;
      description = "FQDN for the Kanboard instance.";
      example = "kanboard.example.org";
    };

    nginx = mkOption {
      type = types.nullOr (types.submodule
        (import ../web-servers/nginx/vhost-options.nix {
          inherit config lib;
        }));
      default = null;
      description = ''
        With this option, you can customize an NGINX virtual host which already has sensible defaults for Kanboard.
        Set to `{ }` if you do not need any customization for the virtual host.
        If enabled, then by default, the {option}`serverName` is
        `''${domain}`,
        If this is set to null (the default), no NGINX virtual host will be configured.
      '';
      example = literalExpression ''
        {
          enableACME = true;
          forceHttps = true;
        }
      '';
    };
```

We can re-use some of the NGINX module's options here